nodejs/lib/inference-session-impl.ts

```diff
@@ -0,0 +1,222 @@
+import {binding, Binding} from './binding';
+import {InferenceSession as InferenceSessionInterface, InferenceSessionFactory} from './inference-session';
+import {OnnxValue} from './onnx-value';
+import {Tensor} from './tensor';
+
+type SessionOptions = InferenceSessionInterface.SessionOptions;
+type RunOptions = InferenceSessionInterface.RunOptions;
+type FeedsType = InferenceSessionInterface.FeedsType;
+type FetchesType = InferenceSessionInterface.FetchesType;
+type ReturnType = InferenceSessionInterface.ReturnType;
+
+class InferenceSession implements InferenceSessionInterface {
+  constructor(session: Binding.InferenceSession) {
+    this.#session = session;
+    // cache metadata
+    this.inputNames = this.#session.inputNames;
+    this.outputNames = this.#session.outputNames;
+  }
+  run(feeds: FeedsType, options?: RunOptions): Promise<ReturnType>;
+  run(feeds: FeedsType, fetches: FetchesType, options?: RunOptions): Promise<ReturnType>;
+  run(feeds: FeedsType, arg1?: FetchesType|RunOptions, arg2?: RunOptions): Promise<ReturnType> {
+    const fetches: {[name: string]: OnnxValue|null} = {};
+    let options: RunOptions = {};
+    // check inputs
+    if (typeof feeds !== 'object' || feeds === null || feeds instanceof Tensor || Array.isArray(feeds)) {
```

Comment:
What if we get a NonTensor type in feeds (which is currently not supported) ?

Reply:
a typemismatch TypeError will be thrown.

This behavior may be changed once we support NonTensor in next release